File: algorithms/ar-bsrt/python3/ar_bsrt.py
def bubble_sort(arr):
    for i in range(len(arr)):
        for j in range(i + 1, len(arr)):
            if arr[i] > arr[j]:
                temp = arr[i]
                arr[i] = arr[j]
                arr[j] = temp
    return arr


if __name__ == "__main__":
    print(bubble_sort([5, 4, 3, 2, 1]))

File: algorithms/ar-bsrt/typescript/ar-bsrt.ts
const bubbleSort = (values: number[]) => {
    const n = values.length;
    const result = [...values];

    for (let i = 0; i < n - 1; i++) {
        for (let j = 0; j < n - 1; j++) {
            if (result[j] > result[j + 1]) {
                const temp = result[j];
                result[j] = result[j + 1];
                result[j + 1] = temp;
            }
        }
    }

    return result;
};

console.log(bubbleSort([23, 42, 4, 15, 8, 16]));

